gitlab-status/handler.go

```diff
@@ -0,0 +1,132 @@
+package function
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/alexellis/hmac"
+	"github.com/openfaas/openfaas-cloud/sdk"
+)
+
+// Handle a serverless request
+func Handle(req []byte) string {
+	validateError := validateRequest(req)
+	if validateError != nil {
+		log.Fatal(validateError)
+	}
+
+	status, statusErr := sdk.UnmarshalStatus(req)
+	if statusErr != nil {
+		return statusErr.Error()
+	}
+
+	token, tokenErr := sdk.ReadSecret("gitlab-api-token")
+	if tokenErr != nil {
+		return tokenErr.Error()
+	}
+
+	url := gitLabURLBuilder(status)
+
+	for _, commitStatus := range status.CommitStatuses {
+		reportErr := sendReport(url, token, commitStatus.Status, commitStatus.Description, commitStatus.Context)
+		if reportErr != nil {
+			log.Fatalf("failed to report status %v, error: %s", status, reportErr.Error())
+		}
+	}
+
+	return ""
+}
+
+func gitLabURLBuilder(status *sdk.Status) string {
+	delimeterPlace := 3
+	delimeterValue := "/"
+	baseURL := getURLbyDelimeter(status.EventInfo.URL, delimeterPlace, delimeterValue)
+	routeURL := fmt.Sprintf("/api/v4/projects/%d/statuses/%s?", status.EventInfo.InstallationID, status.EventInfo.SHA)
+	newURL := append(strings.Split(baseURL, ""), routeURL)
+	wholeURL := strings.Join(newURL, "")
+	return wholeURL
+}
```

Comment:
It should be unit tested

Reply:
Removed the function altogether and remade it added unit test for it